### DISCUSSION TITLE
__sync_val_fetch_and_add未定义

### DISCUSSION THREAD

**Discussion by yinghaoyu** [chenshuo/muduo#613] [Q&A]
我把Atomic.h单独拿出来测试， 提示‘__sync_val_fetch_and_add’ was not declared in this scope; did you mean ‘__sync_fetch_and_add’?
请问为何？g++9.4.0

**Comment by chenshuo** [comment]
What's the output of `g++ -v`?

On Debian 11 x86-64, works fine for me:

```sh
[~/muduo]$ g++ muduo/base/tests/Atomic_unittest.cc -I . -Wall
[~/muduo]$ ./a.out
```

**Reply by yinghaoyu** [reply]
Thanks for your patience & reading.
It's all my fault, as `__sync_fetch_and_add` is the right bulid-in function.
